File: ios/ObjectiveC/Beacon/6-Welcome-Notification/Complete/WelcomeNotification/ViewController.m
//
//  ViewController.m
//  WelcomeNotification
//
//  Created by sarra srairi on 20/12/2016.
//  Copyright © 2016 com.connecthings. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
   // self.txtMessage.text = NSLocalizedString(@"beacon_content_empty", @"");
    
    // Do any additional setup after loading the view, typically from a nib.
    
    [[ATBeaconManager sharedInstance] registerNotificationContentDelegate:self];
    
}

-(void)viewWillAppear:(BOOL)animated{
    //self.txtMessage.text = messageString;
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

//Method to retreive notification BeaconContent from the AppDelegate
- (void)remoteNotificationReceived:(NSNotification *)notification{
    
}

-(void)didReceiveNotificationContentReceived:(ATBeaconContent *)_beaconContent {
    if (_beaconContent) {
        self.txtMessage.text = [_beaconContent getNotificationTitle];
        [self.txtMessage setNeedsDisplay];
    }
}

-(void)didReceiveWelcomeNotificationContentReceived:(ATBeaconWelcomeNotification *)_welcomeNotificationContent {
    
    if (_welcomeNotificationContent) {
        self.txtMessage.text = _welcomeNotificationContent.description;
        [self.txtMessage setNeedsDisplay];
    }
}

@end

File: ios/SWIFT/Beacon/8-Notification-Strategy/Starter/NotificationStrategy/ViewController.swift
//
//  ViewController.swift
//  NotificationStrategy
//
//  Created by sarra srairi on 11/08/2016.
//  Copyright © 2016 R&D connecthings. All rights reserved.
//
import UIKit
import ATConnectionHttp
import ATAnalytics
import ATLocationBeacon
class ViewController: UIViewController,ATBeaconReceiveNotificatonContentDelegate {
    
    @IBOutlet weak var txtMessage: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()
      ATBeaconManager.sharedInstance().registerNotificationContentDelegate(self);
        // Do any additional setup after loading the view, typically from a nib
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
  
    func didReceiveNotificationContentReceived(_ _beaconContent: ATBeaconContent!) {
        self.txtMessage.text = _beaconContent.getNotificationTitle()
        self.txtMessage.setNeedsDisplay()
    }
    
    func didReceiveWelcomeNotificationContentReceived(_ _welcomeNotificationContent: ATBeaconWelcomeNotification!) {
        
    }
}
